src/pages-and-resources/data/thunks.js

```diff
@@ -60,3 +62,31 @@ export function updateAppStatus(courseId, appId, state) {
     }
   };
 }
+
+export function fetchCourseAppSettings(courseId, settings) {
+  return async (dispatch) => {
+    dispatch(updateLoadingStatus({ status: RequestStatus.IN_PROGRESS }));
+
+    try {
+      const settingValues = await getCourseAdvancedSettings(courseId, settings);
+      dispatch(fetchCourseAppsSettingsSuccess(settingValues));
+      dispatch(updateLoadingStatus({ status: RequestStatus.SUCCESSFUL }));
+    } catch (error) {
+      dispatch(updateLoadingStatus({ status: RequestStatus.FAILED }));
+    }
+  };
+}
+
```

Comment:
My question here is what if the API calls fail what is the intended behavior, what should be visible to the user. 
I am raising this as I faced an issue in which API returns 404 the component is calling the API action infinitely from the component and it does not look good. That should be properly handled.

Reply:
The piece of code which is calling this function infinite times on API failure is already merged and part of the current code base. https://github.com/edx/frontend-app-course-authoring/blob/d5e2dbc94dae704419ea3b8f9979415c84c720ad/src/pages-and-resources/PagesAndResources.jsx#L74

Reply:
That is odd. I did not face an infinite loop here, although I have stress-tested the failure path. Will look into this and update.

Reply:
I've refactored this part of the code to eliminate this issue. 

The problem was that in case of a failed request, the data returned would be null so it would request it again. I've added a check to not do that in case the request is failed or denied. Additionally it will now show the error status in the popup.

Reply:
Great, that looks good.